src/main/java/com/microsoft/aad/msal4j/MsalServiceExceptionFactory.java

```diff
@@ -34,7 +34,7 @@ static MsalServiceException fromHttpResponse(HTTPResponse httpResponse){
                 errorResponse.error().equalsIgnoreCase(AuthenticationErrorCode.INVALID_GRANT)) {
 
             if(isInteractionRequired(errorResponse.subError)){
-                throw new MsalInteractionRequiredException(errorResponse, httpResponse.getHeaderMap());
+                return new MsalInteractionRequiredException(errorResponse, httpResponse.getHeaderMap());
```

Comment:
>return [](start = 16, length = 6)

? - is the exception thrown somewhere else?

Reply:
Yes. MsalServiceExceptionFactory should only create exceptions. This was a bug.